src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2654,6 +2656,61 @@ private Process StartWithShellExecute(ProcessStartInfo startInfo)
         #endregion
     }
 
+    /// <summary>
+    /// Provides argument completion for Verb parameter.
+    /// </summary>
+    public class VerbArgumentCompleter : IArgumentCompleter
+    {
+        private const string StartProcessCommandName = "Start-Process";
+        private const string FilePathParameterName = "FilePath";
+        private const string VerbParameterName = "Verb";
+
+        /// <summary>
+        /// Returns completion results for verb parameter.
+        /// </summary>
+        /// <param name="commandName">The command name.</param>
+        /// <param name="parameterName">The parameter name.</param>
+        /// <param name="wordToComplete">The word to complete.</param>
+        /// <param name="commandAst">The command AST.</param>
+        /// <param name="fakeBoundParameters">The fake bound parameters.</param>
+        /// <returns>List of Completion Results.</returns>
+        public IEnumerable<CompletionResult> CompleteArgument(
+            string commandName,
+            string parameterName,
+            string wordToComplete,
+            CommandAst commandAst,
+            IDictionary fakeBoundParameters)
+        {
+            // -Verb is not supported on non-Windows platforms as well as Windows headless SKUs
+            if (!Platform.IsWindowsDesktop
+                || !parameterName.Equals(VerbParameterName, StringComparison.OrdinalIgnoreCase))
+            {
+                yield break;
+            }
+
+            if (commandName.Equals(StartProcessCommandName, StringComparison.OrdinalIgnoreCase)
+                && fakeBoundParameters.Contains(FilePathParameterName))
+            {
+                string filePath = fakeBoundParameters[FilePathParameterName].ToString();
+
+                if (Path.HasExtension(filePath))
```

Comment:
I think mainly to avoid getting process start info on a file which has no extension. Seemed like a reasonable check to not even enter the code block in the first place.

Reply:
Removed. I think its fine and reduces complexity without having check.

Reply:
Yes, I think we can do the check first.

Reply:
Cool. I have put the check back. It seems better to completely avoid `new ProcessStartInfo` if no extension exists because it would be pointless to try if file has no extension.